python/resdata/resfile/rd_kw.py

```diff
@@ -594,23 +587,19 @@ def __IADD__(self, delta, add=True):
                 else:
                     raise TypeError("Type / size mismatch")
             else:
-                if add:
-                    sign = 1
-                else:
-                    sign = -1
+                sign = 1 if add else -1
 
                 if self.data_type.is_int():
                     if isinstance(delta, int):
                         self._shift_int(delta * sign)
                     else:
                         raise TypeError("Type mismatch")
+                elif isinstance(delta, (int, float)):
+                    self._shift_float(
+                        delta * sign
+                    )  # Will call the _float() or _double() function in the C layer.
                 else:
-                    if isinstance(delta, int) or isinstance(delta, float):
-                        self._shift_float(
-                            delta * sign
-                        )  # Will call the _float() or _double() function in the C layer.
-                    else:
-                        raise TypeError("Type mismatch")
+                    raise TypeError("Type mismatch")
```

Comment:
In the tests, the `elif isinstance(delta, (int, float))` is never false. We should add a tests where this falls through to the following `TypeError` on line 613 in the original code.